### PR TITLE
Add Rotorua lakes council (NZ) source

### PR DESCRIPTION
**Pull request by PMKA**
This Pull request aims to introduce a source for integration with Rotorua Lakes Council rubbish collection service. The integration retrieves rubbish collection information based on a given address using the Rotorua Lakes Councils GIS API.
##### Changes:
- Added rotorua_lakes_council_nz.py
- Updated README.md,
- Added rotorua_lakes_council_nz.md

##### Test Results:

    Testing source rotorua_lakes_council_nz ...
      found 2 entries for Test1
        2024-07-31 : Rubbish and recycling [mdi:recycle]
        2024-08-07 : Rubbish only [mdi:trash-can]
      found 2 entries for Test2
        2024-07-29 : Rubbish and recycling [mdi:recycle]
        2024-08-05 : Rubbish only [mdi:trash-can]
      found 2 entries for Test3
        2024-07-30 : Rubbish only [mdi:trash-can]
        2024-08-06 : Rubbish and recycling [mdi:recycle]
      found 2 entries for Test4
        2024-07-29 : Rubbish and recycling [mdi:recycle]
        2024-08-05 : Rubbish only [mdi:trash-can]
      found 2 entries for Test5
        2024-08-01 : Rubbish and recycling [mdi:recycle]
        2024-08-08 : Rubbish only [mdi:trash-can]

### CODE REVIEW

**Review by 5ila5 — CHANGES_REQUESTED** [review]
You can use the logger by adding 
```python
import logging
_LOGGER = logging.getLogger(__name__)
```
and later log with

```python
_LOGGER.error("test ERROR") # not necessary here
# or
_LOGGER.warning("test Warning")
```

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/rotorua_lakes_council_nz.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import datetime
+import requests
+from waste_collection_schedule import Collection
+from geopy.geocoders import Nominatim
```
[the remark this answers is not in the document]
Comment (PMKA) [reply custom_components/waste_collection_schedule/waste_collection_schedule/source/rotorua_lakes_council_nz.py:4]: Removed geopy dependency, using direct calls to OSM instead

**Comment by PMKA** [comment]
Thanks for the feedback @5ila5, have made the requested changes, let me know if there's anything else needs attention.

**Comment by 5ila5** [comment]
Do you think it would be a good idea to `Rubbish and recycling` into two events and strip the `only` from `Rubbish only`? This would make it easier to make automations on rubbish events as you do not need to define both.

But I'm fine with the current implementation if you prefer this one.

**Review by 5ila5 — APPROVED** [review]

**Comment by PMKA** [comment]
Good call, made the change to simplify the output to be `Rubbish` and `Recycling`.

**Comment by 5ila5** [comment]
That's not really what I meant now you're just ignoring the `Rubbish` part of `Rubbish and recycling` and therefore removing information. I thought about splitting `Rubbish and recycling` up into a `Rubbish` and a `Recycling` Collection at the same day

**Comment by PMKA** [comment]
> That's not really what I meant now you're just ignoring the `Rubbish` part of `Rubbish and recycling` and therefore removing information. I thought about splitting `Rubbish and recycling` up into a `Rubbish` and a `Recycling` Collection at the same day

Misunderstanding there. On days where both bin types are collected, I've now split them into separate outputs, Test results below:

Testing source rotorua_lakes_council_nz ...
  found 3 entries for Test1
    2024-08-07 : Rubbish [mdi:trash-can]
    2024-08-14 : Rubbish [mdi:trash-can]
    2024-08-14 : Recycling [mdi:recycle]
  found 3 entries for Test2
    2024-08-05 : Rubbish [mdi:trash-can]
    2024-08-12 : Rubbish [mdi:trash-can]
    2024-08-12 : Recycling [mdi:recycle]
  found 3 entries for Test3
    2024-08-06 : Rubbish [mdi:trash-can]
    2024-08-06 : Recycling [mdi:recycle]
    2024-08-13 : Rubbish [mdi:trash-can]
  found 3 entries for Test4
    2024-08-05 : Rubbish [mdi:trash-can]
    2024-08-12 : Rubbish [mdi:trash-can]
    2024-08-12 : Recycling [mdi:recycle]
  found 3 entries for Test5
    2024-08-02 : Rubbish [mdi:trash-can]
    2024-08-09 : Rubbish [mdi:trash-can]
    2024-08-09 : Recycling [mdi:recycle]

**Comment by 5ila5** [comment]
Thanks for your contribution. I did some minor changes to the logic and reformatted the file based on .pre-commit-config.yaml, I also run the update_docu_links.py script

**Pull request merged**